### DISCUSSION TITLE
Issue with empty list passed to functions

### DISCUSSION THREAD

**Discussion by sheikhanish** [terraform-linters/tflint#2153] [Q&A]
Hi,

I am getting an error with var.control_plane_subnet_ids as empty list of string,
     var.subnet_ids as empty list of string.
Call to function "coalescelist" failed: no non-null arguments. 

Here is the how we are calling coalescelist and variable declaration of control_plane_subnet_ids

variable "control_plane_subnet_ids" {
  description = "List of subnet IDs where the EKS cluster control plane (ENIs) will be provisioned. Used for expanding the pool of subnets used by nodes/node groups without replacing the EKS control plane"
  type        = list(string)
  default     = []
 
}

variable "subnet_ids" {
  description = "List of subnet IDs where the nodes/node groups will be provisioned. If `control_plane_subnet_ids` is not provided, the EKS cluster control plane (ENIs) will be provisioned in these subnets"
  type        = list(string)
  default     = []
}
control_plane_subnet_ids   = var.network_state_file != null ? coalescelist(local.vpc.private_subnet_ids, local.vpc.snx_subnet_ids) : coalescelist(var.control_plane_subnet_ids, var.subnet_ids)

This error comes cause tflint considers that we might pass empty list to coalescelist function. we do not want to change the the coding logic. Hence we want to suppress only this warning but I am not finding a appropriate role which can do this. 

Can you let me know how to deal with this ?

**Comment by bendrucker** [comment]
The lack of formatting makes this post very hard to read.

If you have `default = []` on every variable ultimately passed to `coalescelist`, your module is not valid by default. 

You can either fix your module so that it's valid with the default variable values or override the defaults so that you have a valid module:

https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/config.md#variables

`tflint-ignore` only supports issues raised by rules. It cannot be used to suppress parsing and expression evaluation errors (which this is).

**Reply by sheikhanish** [reply]
tflint is for static analysis, I am wondering why even it is trying to capture  expression evaluation errors even when there is no rules to evaluate it.
I am not getting the same error if I run terraform validate, but the error is thrown if I run tflint.
Any help why is it  like this ?

**Reply by bendrucker** [reply]
> tflint is for static analysis

Yes, which requires evaluating both the structure of the document and the dynamic expressions therein. 

> when there is no rules to evaluate it.

There is, thus the error.

The output should tell you which rule encountered the evaluation and reading the rule's code would explain why it wants to evaluate the expression.
